src/Files.App.Storage/Storables/WindowsStorage/STATask.cs

```diff
@@ -143,5 +146,46 @@ public static Task Run(Func<Task> func)
 
 			return tcs.Task;
 		}
+
+		public unsafe static Task RunAsSync(Action action)
+		{
+			Debug.Assert(Thread.CurrentThread.GetApartmentState() is ApartmentState.STA);
+
+			HANDLE hEventHandle = PInvoke.CreateEvent((SECURITY_ATTRIBUTES*)null, true, false, default);
+
+			var tcs = new TaskCompletionSource();
+
+			Task.Run(() =>
+			{
+				try
+				{
+					action();
+					tcs.SetResult();
+				}
+				catch (Exception ex)
+				{
+					tcs.SetException(ex);
+				}
+				finally
+				{
+					PInvoke.SetEvent(hEventHandle);
+				}
+			});
+
+			HANDLE* pEventHandles = stackalloc HANDLE[1];
```

Comment:
Do you even need to allocate a stack here? I thought your HANDLE variable is in stack already.

Reply:
Related to your 2nd comment, it looks like you can't take address of a local variable that goes thru an async lambda/function. The current code doesn't have async and initially did so this can be refactored but IIUC if we wanna use async void we may have to do this.